src/controllers/Cart.controller.ts

```diff
@@ -160,3 +162,48 @@ export const clearCart = async (req: Request, res: Response) => {
     }
   }
 };
+
+// Buyer update a cart
+export const updateCartProduct = async (req: Request, res: Response) => {
+  try {
+    const userId = req.user.id;
+    const productId = req.params.id;
+
+    const cartProductToUpdate = await findCartProductService(productId, userId);
+    const parsedCartProductToUpdate = JSON.parse(
+      JSON.stringify(cartProductToUpdate),
+    );
+    if (parsedCartProductToUpdate) {
+      for (const productCriteria in req.body) {
+        if (req.body[productCriteria]) {
+          parsedCartProductToUpdate[productCriteria] =
+            req.body[productCriteria];
```

Comment:
this can be too dangerous, because user can even insert whatever he wants. try joi

Reply:
@ISHIMWE-Aime, there is actually already a middleware for that. Just like @musllim said, head over to the middleware folder and check.

Reply:
Yes I understood